### PR TITLE
adjust not on blocks for scheduler input

### LINKED ISSUE #1959: Don't remember `notOn` information if it is in the past
**Is your feature request related to a problem? Please describe.**
It's very annoying when the app stores irrelevant information, especially if that is also invalid input for the scheduler. Right now, if I postpone a Task in the My Time page, a `notOn` block gets added to the Goal information. When it is sent to the scheduler, it looks something like this:  
```
{
  "startDate": "2024-07-20T00:00:00",
  "endDate": "2024-07-27T00:00:00",
  "goals": [
    {
      "id": "da66aeab-4d6d-46d7-8117-4240895e7044",
      "title": "A Goal for which Tasks were postponed",
      "createdAt": 1711689651172,
      "minDuration": 1,
      "notOn": [
        {
          "start": "2024-05-28T09:00:00",
          "end": "2024-05-29T09:00:00"
        },
        {
          "start": "2024-05-29T09:00:00",
          "end": "2024-05-30T09:00:00"
        }
      ]
    }
  ],
  "tasksCompletedToday": []
}
```
The fact that I don't want the scheduler to schedule this on between 2024-05-28T09:00:00 and 2024-05-30T09:00:00 is clearly irrelevant to the scheduler, as the requested Calendar starts a lot later, on the current day: 2024-07-20T00:00:00   

**Describe the solution you'd like**
The notOn blocks in Goals should be automatically shortened to start on midnight of today, or removed completeley. This check and adjustment should be done before sending the Goals to the scheduler.

**Describe alternatives you've considered**
Let the scheduler ignore any notOn blocks in the past. This fixes the scheduler panicking, but doesn't resolve the fact that the app is hanging on to information that is irrelevant. Also the amount of this information is expected to build up quickly over time as postponing Tasks is a regular thing in daily use.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by vinaybadgujar102**
resolves #1959

### CODE REVIEW

**Review by tijlleenders — APPROVED** [review]
Awesome, this seems to fix my issues. Will have to use it for a while to see if it actually does the right things.

**Comment by tijlleenders** [comment]
@Tushar-4781 Code ok?

File: src/api/TasksAPI/index.ts [hunk 1]
```diff
@@ -168,3 +169,57 @@ export const addBlockedSlot = async (goalId: string, slot: { start: string; end:
     console.log(e.stack || e);
   });
 };
+export const updateBlockedSlotsInDB = async (goalId: string, newBlockedSlots: blockedSlotOfTask[]) => {
+  try {
+    await db.transaction("rw", db.taskCollection, async () => {
```
Comment (Tushar-4781) [inline src/api/TasksAPI/index.ts:174]: remove the transaction its not required

File: src/api/TasksAPI/index.ts [hunk 2]
```diff
@@ -168,3 +169,57 @@ export const addBlockedSlot = async (goalId: string, slot: { start: string; end:
...
+
+  const adjustedGoals = goals.map((goal) => {
+    if (!goal.notOn || goal.notOn.length === 0) {
+      return goal;
+    }
+
+    let isUpdated = false;
+
+    const adjustedNotOn = goal.notOn
+      .filter((block) => {
```
Comment (Tushar-4781) [inline src/api/TasksAPI/index.ts:198]: try this    .filter((block) => new Date(block.end) > today)

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]

**Comment by vinaybadgujar102** [comment]
@Tushar-4781, you can review this

**Review by Tushar-4781 — APPROVED** [review]

**Pull request merged**